File: front/EdTech/src/app/quiz/model/question.ts
export interface Question {
  id?: number;
  questionTitle: string;
  option1: string;
  option2: string;
  option3: string;
  rightAnswer?: string;
  category?: string;
}

export interface QuestionRequest {
  questionTitle: string;
  option1: string;
  option2: string;
  option3: string;
  rightAnswer?: string;
  category?: string;
}

export interface QuestionResponse{
  id: number;
  response: string;
}

File: src/main/java/pi/tn/esprit/models/Question.java
package pi.tn.esprit.models;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "questions")
@Data
public class Question {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String questionTitle;
    private String option1;
    private String option2;
    private String option3;
    private String rightAnswer;
    private String category;
}
